### ISSUE TITLE
integrate(sqrt(2*m*(E - x)), x)

### ISSUE THREAD

**Issue by suhr** [diofant/diofant#842]
![diofant](https://user-images.githubusercontent.com/65870/58827305-61069000-864b-11e9-9f7d-4206cde46c2e.png)

But `integrate(sqrt(2*(E - x)), x)` returns a sane result.

**Comment by skirpichev** [comment]
I'm not sure that `Out[2]` is wrong.  Rather this looks as unsimplified expression.

With `m = Symbol('m', positive=True)` it does return a simple expression as well.

**Comment by skirpichev** [comment]
Minimal test case:
```
In [8]: from diofant.integrals.meijerint import meijerint_indefinite

In [9]: meijerint_indefinite(sqrt(x*y), x)
Out[9]: 
⎧                             3/2   ___                                        
⎪                          2⋅x   ⋅╲╱ y                                         
⎪                          ────────────                             for │x│ < 1
⎪                               3                                              
⎨                                                                              
⎪  ___ ╭─╮1, 1 ⎛ 1   5/2 │  ⎞     ___ ╭─╮0, 2 ⎛5/2, 1         │  ⎞             
⎪╲╱ y ⋅│╶┐     ⎜         │ x⎟ + ╲╱ y ⋅│╶┐     ⎜               │ x⎟   otherwise 
⎪      ╰─╯2, 2 ⎝3/2   0  │  ⎠         ╰─╯2, 2 ⎝        3/2, 0 │  ⎠             
⎩                                                                               
```

I think, the `_split_mul()` helper may be the source of problem (it factors out the `sqrt(y)`).

PS: BTW, please don't use screenshots in bugreports, unless it's unavoidable (e.g. plots). You can copy-paste text directly, as well - from the jupyter console or notebook.

**Comment by skirpichev** [comment]
@suhr, I believe this was fixed by ac3c182e8f.  In the version 0.14.0:
```python
$ python -m diofant

In [1]: var('x m E')
Out[1]: (x, m, E)

In [2]: integrate(sqrt(2*m*(E - x)), x)
Out[2]: 
      ⎛        ___________         ___________⎞
  ___ ⎜  2⋅E⋅╲╱ E⋅m - m⋅x    2⋅x⋅╲╱ E⋅m - m⋅x ⎟
╲╱ 2 ⋅⎜- ───────────────── + ─────────────────⎟
      ⎝          3                   3        ⎠

In [3]: simplify(_)
Out[3]: 
    ___   ___________         
2⋅╲╱ 2 ⋅╲╱ m⋅(E - x) ⋅(-E + x)
──────────────────────────────
              3               
```
The answer from ``meijerint_indefinite()`` also seems to be correct.

So, I'll close this issue.
